core-bundle/src/Resources/contao/library/Contao/Controller.php

```diff
@@ -367,7 +368,7 @@ public static function getFrontendModule($intId, $strColumn='main')
 		}
 
 		// Check the visibility (see #6311)
-		if (!static::isVisibleElement($objRow))
+		if (TL_MODE === 'FE' && !System::getContainer()->get('security.helper')->isGranted(AbstractFrontendAccessVoter::ATTRIBUTE, $objRow))
```

Comment:
Shouldn't the scope check be part of the `AbstractFrontendAccessVoter::voteOnAttribute()` method? Because any front end voter should deny if in back end scope.

Reply:
> I don't think it's correct for voters to check on request attributes. What should be voted on and what not is defined by the attributes, not any scope.

The voter does not know if you're in front end or back end. Nor should it behave differently depending on the firewall.